apps/projects/routes.py

```diff
@@ -1093,17 +1198,32 @@ def _get_timeline_thumbnails(self, amount):
             )
             return json_response({"processing": True}, status=202)
 
-    def _get_preview_thumbnail(self, position):
+    def _get_preview_thumbnail(self, position, crop, rotate):
         """
         Get or create thumbnail for preview
         :param position: video position to capture a frame
         :type position: int
+        :param crop: crop editing rules
+        :type crop: dict
+        :param rotate: rotate degree
+        :type rotate: int
         :return: json response
         :rtype: flask.wrappers.Response
         """
+        # validate crop param
+        if crop:
+            if self.project['metadata']['width'] - crop['x'] < app.config.get('MIN_VIDEO_WIDTH'):
+                raise BadRequest({"crop": [{"x": ["less than minimum allowed crop width"]}]})
+            elif self.project['metadata']['height'] - crop['y'] < app.config.get('MIN_VIDEO_HEIGHT'):
+                raise BadRequest({"crop": [{"y": ["less than minimum allowed crop height"]}]})
+            elif crop['x'] + crop['width'] > self.project['metadata']['width']:
+                raise BadRequest({"crop": [{"width": ["crop's frame is outside a video's frame"]}]})
+            elif crop['y'] + crop['height'] > self.project['metadata']['height']:
+                raise BadRequest({"crop": [{"height": ["crop's frame is outside a video's frame"]}]})
+
         # resource is busy
         if self.project['processing']['thumbnail_preview']:
-            return json_response({"processing": True}, status=409)
+            raise Conflict({"processing": ["Task get preview thumbnails video is still processing"]})
         elif (self.project['thumbnails']['preview'] and
               self.project['thumbnails']['preview'].get('position') == position):
```

Comment:
Changes should affect also this part. If we have a preview thumbnail on position `5` we can't crop it using the same position, it will return current thumbnail data.

Reply:
I think we should drop checking position. This case is hardly happen, in reality client can't capture same time (always has some millisecond difference), so it does not save us much, and it affect to crop and rotate in preview thumbnail.

Reply:
@idsvn if client will not request for the same thumbnail (the same position) each time editor is inited - it's fine to remove this condition.